packages/neuron-wallet/tests/services/tx/transaction-service.test.ts

```diff
@@ -0,0 +1,243 @@
+import { TransactionStatus } from '../../../src/models/chain/transaction'
+import { saveTransactions, loadTransactions, initConnection, closeConnection, saveAccounts } from '../../setupAndTeardown'
+
+const transactions = loadTransactions()
+
+describe('Test transaction service', () => {
+  beforeEach(async () => {
+    await initConnection()
+    await saveAccounts()
+    return saveTransactions(transactions)
+  })
+
+  afterEach(() => {
+    return closeConnection()
+  })
+
+  describe('Test filter search type', () => {
+    it('Should return empty', () => {
+      const actual = TransactionService.filterSearchType('')
+      expect(actual).toBe(SearchType.Empty)
+    })
+
+    it('Should return address', () => {
+      const MAINNET_ADDRESS = 'ckb1qyqv9w4p6k695wkkg54eex9d3ckv2tj3y0rs6ctv00'
+      const TESTNET_ADDRESS = 'ckt1qyqv9w4p6k695wkkg54eex9d3ckv2tj3y0rs6ctv00'
+      const ADDRESSES = [MAINNET_ADDRESS, TESTNET_ADDRESS]
+      ADDRESSES.forEach(addr => {
+        expect(TransactionService.filterSearchType(addr)).toBe(SearchType.Address)
+      })
+    })
+
+    it('Should return transaction hash', () => {
+      const TX_HASH = '0x884c79635976a09d1bee84a4bbcc19454cbeb05e831b1d87cb39e20c73e63833'
+      const actual = TransactionService.filterSearchType(TX_HASH)
+      expect(actual).toBe(SearchType.TxHash)
+    })
+
+    it('Should return date', () => {
+      const DATE = '2019-02-09'
+      const actual = TransactionService.filterSearchType(DATE)
+      expect(actual).toBe(SearchType.Date)
+    })
+
+    it('Should return token info if no value matched', () => {
+      const VALUE = 'unknown'
+      const actual = TransactionService.filterSearchType(VALUE)
+      expect(actual).toBe(SearchType.TokenInfo)
+    })
+  })
+
+  it('Test blake160sOfTx', () => {
+    const res = TransactionService.blake160sOfTx(transactions[0])
+    expect(res).toEqual([
+      "0x36c329ed630d6ce750712a477543672adab57f4c",
+      "0xe2193df51d78411601796b35b17b4f8f2cd85bd0"
+    ])
+  })
+
+  describe('Test getCountByLockHashesAndStatus', () => {
+    it('Should return result', async () => {
+      const lockHash = '0x1f2615a8dde4e28ca736ff763c2078aff990043f4cbf09eb4b3a58a140a0862d'
+      const lockHashList = new Set([lockHash])
+      const status = new Set([TransactionStatus.Success])
+      const actual = await TransactionService.getCountByLockHashesAndStatus(lockHashList, status)
+      expect(actual).toEqual(new Map([[lockHash, 2]]))
+    })
+    it('Should return empty', async () => {
+      const lockHash = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+      const lockHashList = new Set([lockHash])
+      const status = new Set([TransactionStatus.Success])
+      const actual = await TransactionService.getCountByLockHashesAndStatus(lockHashList, status)
+      expect(actual).toEqual(new Map())
+    })
+  })
+
+  describe('Test update transaction description', () => {
```

Comment:
If it is mainly for testing a function, how about we name the scope as `describe('#updateDescription')`, since the function name is expressive enough and it is easier to locate the tests for a specific function?

Reply:
Updated in https://github.com/nervosnetwork/neuron/pull/1733/commits/00d73419d36f406d3eed6753679f1de1b07055db